services/api/service.go

```diff
@@ -1595,6 +1604,185 @@ func (api *RelayAPI) checkBuilderEntry(w http.ResponseWriter, log *logrus.Entry,
+func (api *RelayAPI) checkTxAndSenderValidity(txs []*types.Transaction) error {
+	// TODO - Payouts still need to be modelled
+	// TODO - expand state interference checks as in checkTobTxsStateInterference using call tracing
+
+	if len(txs) == 0 {
+		return fmt.Errorf("Empty TOB tx request sent!")
+	}
+	if len(txs) == 1 {
+		return fmt.Errorf("We require a payment tx along with the TOB txs!")
+	}
+
+	// Start: Payout checks
+	lastTx := txs[len(txs)-1]
+
+	if lastTx.To() != nil && *lastTx.To() != api.relayerPayoutAddress {
+		return fmt.Errorf("We require a payment tx to the relayer along with the TOB txs!")
+	}
+	if lastTx.Value().Cmp(big.NewInt(0)) == 0 {
+		return fmt.Errorf("The relayer payment tx is non-zero!")
+	}
+	if len(lastTx.Data()) != 0 {
+		return fmt.Errorf("The relayer payment tx has malformed data!")
+	}
+	// TODO - extend payout checks
+	// End: Payout checks
+
+	// State interference checks
+	return api.checkTobTxsStateInterference(txs)
+}
+
+func (api *RelayAPI) handleSubmitNewTobTxs(w http.ResponseWriter, req *http.Request) {
+	headSlot := api.headSlot.Load()
+	receivedAt := time.Now().UTC()
+
+	log := api.log.WithFields(logrus.Fields{
+		"method":                "submitTobTxs",
+		"contentLength":         req.ContentLength,
+		"headSlot":              headSlot,
+		"timestampRequestStart": receivedAt.UnixMilli(),
+	})
+
+	defer func() {
+		log.WithFields(logrus.Fields{
+			"timestampRequestFin": time.Now().UTC().UnixMilli(),
+			"requestDurationMs":   time.Since(receivedAt).Milliseconds(),
+		}).Info("request finished")
+	}()
+
+	tobTxRequest := new(common.TobTxsSubmitRequest)
+
+	var r io.Reader = req.Body
+	limitReader := io.LimitReader(r, 10*1024*1024) // 10 MB
+	requestBytes, err := io.ReadAll(limitReader)
+	if err != nil {
+		log.WithError(err).Warn("could not read payload")
+		api.RespondError(w, http.StatusBadRequest, err.Error())
+		return
+	}
+
+	// Check for JSON encoding
+	contentType := req.Header.Get("Content-Type")
+	if contentType == "application/json" {
+		log = log.WithField("reqContentType", "json")
+		if err := json.Unmarshal(requestBytes, tobTxRequest); err != nil {
+			log.WithError(err).Warn("could not decode payload - JSON")
+			api.RespondError(w, http.StatusBadRequest, err.Error())
+			return
+		}
+	} else {
+		log.Error("We do not support non-JSON requests yet!", contentType)
+		api.Respond(w, http.StatusBadRequest, "invalid content-type")
+		return
+	}
+
+	slot := tobTxRequest.Slot
+	parentHash := tobTxRequest.ParentHash
+	if len(tobTxRequest.TobTxs.Transactions) == 0 {
+		log.Error("Empty TOB tx request sent!")
+		api.Respond(w, http.StatusBadRequest, "Empty TOB tx request sent!")
+		return
+	}
+	//
+	if slot < headSlot {
+		log.Error("TOB tx request for past slot!")
+		api.Respond(w, http.StatusBadRequest, "Submitted TOB tx request for past slot!")
+		return
+	}
+
+	// We only allow bidding for TOB 1 slot prior
```

Comment:
add punctuation to comments. here and below

Reply:
sounds good!